File: python/BubbleSort.py
import time

class BubbleSort:
    def __init__(self, myList):
        self.myList = myList

    def sort(self, wait_time=0.02):
        start = time.time()
        listSorted = False

        while not listSorted:
            check = True
            for i in range(len(self.myList) - 1):
                time.sleep(wait_time)
                if self.myList[i] > self.myList[i + 1]:
                    self.myList[i], self.myList[i + 1] = self.myList[i + 1], self.myList[i]
                    check = False
                    self.visualizator.drawList(self.myList)
                    
            if check:
                listSorted = True

        end = time.time()
        print(end - start)

File: BubbleSort.cpp
#include <chrono>
#include <iostream>
#include "BubbleSort.h"

void BubbleSort::sort(){
    typedef std::chrono::milliseconds ms;
    typedef std::chrono::duration<float> fsec;

    auto start = std::chrono::system_clock::now();
    bool listSorted = false;

    int temp;

    while (!listSorted){
        bool check = true;
        for (int i = 0; i < sortArraySize - 1; i++){
            if (sortArray[i] > sortArray[i+1]){
                std::swap(sortArray[i], sortArray[i+1]);
                check = false;

                for(int i = 0; i < sortArraySize; i++){
                    std::cout << sortArray[i] << " ";
                }
                std::cout << "\n";
            }    
        }
        if (check) listSorted = true;
    }

    auto end = std::chrono::system_clock::now();
    fsec time = end - start;
    ms d = std::chrono::duration_cast<ms>(time);
    
    std::cout << d.count() << "ms\n";
}
